### PR TITLE
Jmm/audit period other

### PR DESCRIPTION
**Pull request by JeanMarie-TTS**

### CODE REVIEW

File: backend/report_submission/forms.py
Comment (danswick) [inline backend/report_submission/forms.py:13]: I believe we need to add the new field here

File: backend/report_submission/templates/report_submission/gen-form.html
Comment (danswick) [inline backend/report_submission/templates/report_submission/gen-form.html:139]: This change adds an empty string as a default value and only disables the form if that value is empty. I _think_ the always-on `disabled` state was clobbering the input value before.

File: backend/schemas/source/sections/GeneralInformation.schema.jsonnet
Comment (danswick) [inline backend/schemas/source/sections/GeneralInformation.schema.jsonnet:28]: Just an auto-formatting gremlin.

Comment (jadudm) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet:28]: If we don't know what the max should be, we shouldn't constrain it. Could it be 24 (biennial)? More? Less?

File: backend/schemas/source/sections/GeneralInformation.schema.jsonnet
Comment (jadudm) [inline backend/schemas/source/sections/GeneralInformation.schema.jsonnet:25]:
```suggestion
    audit_period_covered: Base.Enum.AuditPeriod,
```

Comment (jadudm) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet:25]: Because JSonnet is a compiler, we don't have to use the `$defs` notation to break things out. Or, we should decide what we want our conventions to be. It might be that GeneralInfo used this format, but I think most of the other workbooks/forms just directly reference the typing information. 

It isn't strictly worth it at this point to make this change, but I thought I'd mention it.

File: backend/report_submission/views.py
Comment (jadudm) [inline backend/report_submission/views.py:179]: I'm not sure why `print` statements were brought back into the commit. They could be a `logger.debug`, but they shouldn't stay in as we loko to merge. (Although, I just noticed this is a draft PR.)

File: backend/schemas/source/sections/GeneralInformation.schema.jsonnet
Comment (jadudm) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet:28]: From the Census docs on this in the past...

```
Item 3 - Audit Period Covered
Mark the appropriate box. Annual audits cover 12 months and
biennial audits cover 24 months. If the audit period covered is
neither annual nor biennial, mark “Other” and provide the number
of months covered in the space provided. The number of months
must be entered as a two-digit number (e.g., for three months
enter ‘03’). See 2 CFR 200.504 for guidelines about the
frequency of audits.
```

https://facides.census.gov/Files/2022%20Checklist%20Instructions%20and%20Form.pdf

Which... doesn't help a lot. 

If we do set min/max, I think they're `minimum`, `maximum`, `exclusiveMinimum`, and `exclusiveMaximum`. 

https://json-schema.org/understanding-json-schema/reference/numeric.html

Comment (jadudm) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet:28]: In 2022, it looks like the max was 18 months.

File: backend/schemas/source/sections/GeneralInformation.schema.jsonnet
Comment (JeanMarie-TTS) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet:25]: @danswick , can you address the jsonnet change. There's this comment from @jadud as well as the clarification that audit_period_other_months should be a two-digit string with leadig zeros if not nu..
I am not very familiar with jsonnet and would prefer to get some help here.

Comment (danswick) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet:25]: Let's handle this as a separate task and just revise all of the `$defs` usage in one chunk of work. I added a task to the [Tech Debt epic](https://github.com/GSA-TTS/FAC/issues/1712) and linked back here for later!

Comment (JeanMarie-TTS) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet:25]: In that case, this PR is ready to go.

File: backend/audit/models.py
Comment (jadudm) [inline backend/audit/models.py:1]: @tadhg-ohiggins and I are looking at this.

1. What does this have to do with the audit period merge?
2. The server is UTC. `date.today()` is therefore UTC. Is... this a force on UTC for a reason (e.g. local dev?), or something else? (And, if it is for correctness *on the server*, then should there be a separate PR where we walk the entire codebase for *every* timestamp?)

File: backend/report_submission/views.py
Comment (jadudm) [inline backend/report_submission/views.py:1]: At the point that we're merging, we should probably always remove commented out `print` statements.

File: backend/schemas/output/sections/GeneralInformation.schema.json
Comment (jadudm) [inline backend/schemas/output/sections/GeneralInformation.schema.json:135]: I had comments/questions here on the draft. Do we know why it is 18 months? That needs to be answered. Did @Carley-Sullivan or someone with more insight answer conclusively what the upper bound on this should be? I, at the moment, do not know.

And, could we use either `maximum` or `exlusiveMaximum` per the docs?

https://json-schema.org/understanding-json-schema/reference/numeric.html#range

File: backend/schemas/source/sections/GeneralInformation.schema.jsonnet
Comment (jadudm) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet:28]: Ah. I left my comment on the JSON file. Could you please see that comment. My statement  that 2022 data only had values up to 18 months is not a conclusive answer to what the upper bound in the validation should, or should not, be.

File: backend/Makefile
Comment (jadudm) [inline backend/Makefile:1]: @tadhg-ohiggins points out that this might want to be part of a separate PR? It changes the makefile, which... isn't really related to audit_period_other changes?

**Review by jadudm — CHANGES_REQUESTED** [review]
Questions and comments were left inline.

File: backend/audit/models.py
Comment (JeanMarie-TTS) [reply backend/audit/models.py:1]: This change was to address an annoying Warning, which saod something to the effect that the timezone was ambiguous in date.today()
Not related to this issue, I agree. But it's good to clean up when you see issues in the  code that you are touching and testing.

File: backend/Makefile
Comment (JeanMarie-TTS) [reply backend/Makefile:1]: Thw schema definitions will need to be addressed in a separate PR. This PR has enough to allow the field to be captured and flow all the way through etl.

File: backend/report_submission/test_views.py
Comment (danswick) [inline backend/report_submission/test_views.py:387]:
```suggestion
```

File: backend/report_submission/test_views.py
Comment (danswick) [inline backend/report_submission/test_views.py:390]: Should this line be deleted? 

```suggestion
```

File: backend/report_submission/views.py
Comment (danswick) [inline backend/report_submission/views.py:40]:
```suggestion
```

File: backend/report_submission/views.py
Comment (danswick) [inline backend/report_submission/views.py:102]:
```suggestion
```

File: backend/report_submission/views.py
Comment (danswick) [inline backend/report_submission/views.py:179]:
```suggestion
```

File: backend/schemas/source/sections/GeneralInformation.schema.jsonnet
Comment (danswick) [inline backend/schemas/source/sections/GeneralInformation.schema.jsonnet:28]:
```suggestion
      minimum: 0,
      maximum: 19, 
```

**Review by danswick — COMMENTED** [review]
Added some suggestions for print statements, stray comments, and that `max` -> `maximum` jsonschema thing.

File: backend/schemas/source/sections/GeneralInformation.schema.jsonnet
Comment (danswick) [reply backend/schemas/source/sections/GeneralInformation.schema.jsonnet]: No longer relevant now that we're switching to 2-character strings. (Per uniform guidance, must be in format, "05" etc)

**Review by danswick — APPROVED** [review]
I think we're ready to merge once tests pass.

File: backend/schemas/output/sections/GeneralInformation.schema.json
Comment (jadudm) [inline backend/schemas/output/sections/GeneralInformation.schema.json:135]:
```suggestion
         "pattern": "^0[0-9]|1[0-8]$",
```

File: backend/schemas/source/base/Base.libsonnet
Comment (jadudm) [inline backend/schemas/source/base/Base.libsonnet:296]:
```suggestion
local REGEX_MONTHS_OTHER = '^0[0-9]|1[0-8]$';
```

**Review by jadudm — APPROVED** [review]
LGTM.

**Pull request merged**
